reflect.go

```diff
@@ -310,12 +322,26 @@ func (r *Reflector) reflectStruct(definitions Definitions, t reflect.Type) *Type
 
 func (r *Reflector) reflectStructFields(st *Type, definitions Definitions, t reflect.Type) {
 	t = getNonPointerType(t)
+	// orderArray to determine the order of the reflection for the struct
+	orderArray := []StructOrder{}
 	for i := 0; i < t.NumField(); i++ {
 		f := t.Field(i)
+		// if not Anonymous and tag not present, insert
+		// else check if present and assign propert to current field
+		if !f.Anonymous {
+			_, ok := st.tagPrecedence[f.Name]
```

Comment:
OK :+1:, considering the logic I think this is the cleanest way of writing it

Reply:
Thanks, Wayne. Sure, I will make the changes accordingly.